File: ios/Classes/models/NFCInfoMessages.swift
import Foundation
import FaturamatikVerify

public struct NFCInfoMessages {
    public var tagError: String?
    public var moreThanOneTag: String?
    public var connectionError: String?
    public var invalidMrzKey: String?
    public var responseError: String?
    public var defaultMessage: String?
    public var successMessage: String?
    public var requestPresentPass: String?

    public init(dict: [String: String]) {
        self.tagError = dict["tagError"]
        self.moreThanOneTag = dict["moreThanOneTag"]
        self.connectionError = dict["connectionError"]
        self.invalidMrzKey = dict["invalidMrzKey"]
        self.responseError = dict["responseError"]
        self.defaultMessage = dict["defaultMessage"]
        self.successMessage = dict["successMessage"]
        self.requestPresentPass = dict["requestPresentPass"]
    }

    public func toNFCMessageMap() -> [NFCMessageKey: String] {
        var map: [NFCMessageKey: String] = [:]

        if let tagError = tagError { map[.tagError] = tagError }
        if let moreThanOneTag = moreThanOneTag { map[.moreThanOneTag] = moreThanOneTag }
        if let connectionError = connectionError { map[.connectionError] = connectionError }
        if let invalidMrzKey = invalidMrzKey { map[.invalidMrzKey] = invalidMrzKey }
        if let responseError = responseError { map[.responseError] = responseError }
        if let defaultMessage = defaultMessage { map[.defaultMessage] = defaultMessage }
        if let successMessage = successMessage { map[.successMessage] = successMessage }
        if let requestPresentPass = requestPresentPass { map[.requestPresentPass] = requestPresentPass }

        return map
    }
}

File: lib/common/models/ios/nfc_info_messages.dart
class NFCInfoMessages {
  final String? tagError;
  final String? moreThanOneTag;
  final String? connectionError;
  final String? invalidMrzKey;
  final String? responseError;
  final String? defaultMessage;
  final String? successMessage;
  final String? requestPresentPass;

  NFCInfoMessages({
    this.tagError,
    this.moreThanOneTag,
    this.connectionError,
    this.invalidMrzKey,
    this.responseError,
    this.defaultMessage,
    this.successMessage,
    this.requestPresentPass,
  });

  Map<String, String> toMap() {
    final map = <String, String>{};
    if (tagError != null) map['tagError'] = tagError!;
    if (moreThanOneTag != null) map['moreThanOneTag'] = moreThanOneTag!;
    if (connectionError != null) map['connectionError'] = connectionError!;
    if (invalidMrzKey != null) map['invalidMrzKey'] = invalidMrzKey!;
    if (responseError != null) map['responseError'] = responseError!;
    if (defaultMessage != null) map['defaultMessage'] = defaultMessage!;
    if (successMessage != null) map['successMessage'] = successMessage!;
    if (requestPresentPass != null) map['requestPresentPass'] = requestPresentPass!;
    return map;
  }
}